twoliter/embedded/rpm2kit

```diff
@@ -30,3 +35,90 @@ done
 
 createrepo_c "${KIT_DIR}"
 dnf --disablerepo '*' --repofrompath "kit,file:///${KIT_DIR}" repoquery --all
+
+WORK_DIR=$(mktemp -d)
+TIMESTAMP=$(date +"%FT%T.%NZ")
+FILENAME_PREFIX="${KIT_NAME:?}--${ARCH:?}--${VERSION_ID:?}--${BUILD_ID:?}"
+
+# Create the content layer
+mkdir -p "${WORK_DIR}/blobs/sha256"
+tar -cf "${WORK_DIR}/content-layer.tar" -C "${KIT_DIR}" .
+DIFF_ID=$(digest_from_file "${WORK_DIR}/content-layer.tar")
+gzip "${WORK_DIR}/content-layer.tar"
+CONTENT_DIGEST=$(digest_from_file "${WORK_DIR}/content-layer.tar.gz")
+mv "${WORK_DIR}/content-layer.tar.gz" "${WORK_DIR}/blobs/sha256/${CONTENT_DIGEST}"
+
+# Create the OCI config
+METADATA=$(base64 < "${FILENAME_PREFIX}.json")
+CONFIG=<<EOF
+{
+  "architecture": "${ARCH}",
+  "config": {
+    "Env": [],
+    "WorkingDir": "/",
+    "OnBuild": null,
+    "Labels": {
+      "dev.bottlerocket.kit.v1": "${METADATA}"
+    }
+  },
+  "created": "${TIMESTAMP}",
+  "history": [],
+  "os": "linux",
+  "rootfs": {
+    "type": "layers",
+    "diff_ids": [
+      "sha256:${DIFF_ID}"
+    ]
+  }
+}
+EOF
+CONFIG_DIGEST=$(create_json_layer "${WORK_DIR}" "${CONFIG}")
```

Comment:
`create_json_layer` has a weird contract for a function - it writes out a file and also emits output to be captured - so it might be more obvious just to do this:
```suggestion
CONFIG_DIGEST="$(digest_from_blob "${CONFIG}")"
echo "${CONFIG}" > "${WORK_DIR}/blobs/sha256/${CONFIG_DIGEST}"
```